src/deepsparse/pipeline.py

```diff
@@ -406,17 +419,29 @@ def __call__(self, *args, **kwargs):
             inference_state.create_state({})
             inference_state.set_timer(timer)
 
-        kwargs["inference_state"] = inference_state
+            timer = self.timer_manager.get_new_timer()
+            inference_state.set_timer(timer)
+            is_nested = False
 
         next_call = self.run
         if self.middleware_manager is not None:
+            # make next calls to be middlewares if any
             next_call = self.middleware_manager.build_middleware_stack(next_call)
 
-        rtn = next_call(*args, **kwargs)
+        kwargs["inference_state"] = inference_state
+        # add name for timer measurements key
+        kwargs["name"] = "total"
+        kwargs["is_nested"] = is_nested
 
         # timer shared across all operators, has all measurements
         timer = inference_state.timer
 
+        next_call = self.run
+        if self.middleware_manager is not None:
```

Comment:
these lines repeat what we have on line 427 - is that intended?
Also, it seems like some of the steps being done here are not done in the `aysnc` pathway?

Reply:
Good catch. One is a duplicate.

Yes async pathway does not have a `total`. The entrypoint is run_async -> trigger operators, whereas regular entrypoint is __call__ -> run -> trigger ops, so can wrap in __call__.

Reply:
Does that mean the async pathway wont be able store timings?